addons/mail/static/src/core/web/chatter.xml

```diff
@@ -94,7 +94,7 @@
                 <t t-if="props.hasActivities and activities.length and !state.isSearchOpen">
                     <t t-call="mail.ActivityList"/>
                 </t>
-                <div t-if="state.isAttachmentBoxOpened" class="o-mail-AttachmentBox position-relative" t-ref="attachment-box">
+                <div t-if="state.isAttachmentBoxOpened and attachments.length" class="o-mail-AttachmentBox position-relative" t-ref="attachment-box">
```

Comment:
`isAttachmentBoxOpened` should be updated properly when `attachments.length` is 0. It ensures a consistent behaviour wherever `isAttachmentBoxOpened` is uses.